Description: Ensure .note.GNU-stack is in all asm files to prevent executable stack
Forwarded: https://github.com/01org/isa-l/pull/5
Author: Ondřej Nový <onovy@debian.org>
Last-Update: 2016-06-25

--- a/igzip/bitbuf2.asm
+++ b/igzip/bitbuf2.asm
@@ -28,6 +28,7 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
 %include "options.asm"
+%include "reg_sizes.asm"
 
 ; Assumes m_out_buf is a register
 ; Clobbers RCX
--- a/igzip/huffman.asm
+++ b/igzip/huffman.asm
@@ -28,6 +28,7 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
 %include "options.asm"
+%include "reg_sizes.asm"
 %include "lz0a_const.asm"
 
 ; Macros for doing Huffman Encoding
--- a/igzip/igzip_body.asm
+++ b/igzip/igzip_body.asm
@@ -28,6 +28,7 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
 %include "options.asm"
+%include "reg_sizes.asm"
 %ifndef TEST
 
 extern fold_4
--- a/igzip/igzip_buffer_utils_01.asm
+++ b/igzip/igzip_buffer_utils_01.asm
@@ -31,6 +31,7 @@
 %define	BUFFER_UTILS
 
 %include "options.asm"
+%include "reg_sizes.asm"
 
 extern pshufb_shf_table
 extern mask3
--- a/igzip/igzip_buffer_utils_04.asm
+++ b/igzip/igzip_buffer_utils_04.asm
@@ -31,6 +31,7 @@
 %define	BUFFER_UTILS
 
 %include "options.asm"
+%include "reg_sizes.asm"
 
 extern pshufb_shf_table
 extern mask3
--- a/igzip/igzip_compare_types.asm
+++ b/igzip/igzip_compare_types.asm
@@ -28,6 +28,7 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
 %include "options.asm"
+%include "reg_sizes.asm"
 %ifndef UTILS_ASM
 %define UTILS_ASM
 ; compare macro
--- a/igzip/data_struct2.asm
+++ b/igzip/data_struct2.asm
@@ -27,6 +27,8 @@
 ;  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+%include "reg_sizes.asm"
+
 ;; START_FIELDS
 %macro START_FIELDS 0
 %assign _FIELD_OFFSET 0
--- a/igzip/stdmac.asm
+++ b/igzip/stdmac.asm
@@ -27,6 +27,7 @@
 ;  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+%include "reg_sizes.asm"
 
 ;; internal macro used by push_all
 ;; push args L to R
--- a/igzip/lz0a_const.asm
+++ b/igzip/lz0a_const.asm
@@ -27,6 +27,8 @@
 ;  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+%include "reg_sizes.asm"
+
 %assign K      1024
 %assign D      HIST_SIZE * K        ;; Amount of history
 %assign LA     17 * 16      ;; Max look-ahead, rounded up to 32 byte boundary
--- a/igzip/options.asm
+++ b/igzip/options.asm
@@ -29,6 +29,8 @@
 
 default rel
 
+%include "reg_sizes.asm"
+
 %ifndef __OPTIONS_ASM__
 %define __OPTIONS_ASM__
 
--- a/igzip/crc_data.asm
+++ b/igzip/crc_data.asm
@@ -1,3 +1,5 @@
+%include "reg_sizes.asm"
+
 %ifndef CRC_DATA
 
 %define CRC_DATA
--- a/igzip/igzip_body_01.asm
+++ b/igzip/igzip_body_01.asm
@@ -1,3 +1,5 @@
+%include "reg_sizes.asm"
+
 %define ARCH 01
 
 %ifndef COMPARE_TYPE
--- a/igzip/igzip_body_04.asm
+++ b/igzip/igzip_body_04.asm
@@ -1,3 +1,5 @@
+%include "reg_sizes.asm"
+
 %define ARCH 04
 %define USE_HSWNI
 
--- a/igzip/igzip_stateless_01.asm
+++ b/igzip/igzip_stateless_01.asm
@@ -1,3 +1,5 @@
+%include "reg_sizes.asm"
+
 %define ARCH 01
 
 %ifndef COMPARE_TYPE
--- a/igzip/igzip_stateless_04.asm
+++ b/igzip/igzip_stateless_04.asm
@@ -1,3 +1,5 @@
+%include "reg_sizes.asm"
+
 %define ARCH 04
 %define USE_HSWNI
 
